Polyculture Use Case: Map Specific Favorite Groups (#168)
ElektraInitiative/PermaplanT: doc/usecases/polyculture/map_specific_favorite_groups.md

> # Use Case: Map Specific Favorite Groups

Reviewer: ```suggestion
# Use Case: Map-Specific Favorite Groups
```

## Summary

- **Scope:** Plants Layer
- **Level:** User Goal
- **Actors:** App User
- **Brief:** The user change a set of favorite plant groups per map.

> - **Status:** Draft

Reviewer: Please change status and assign yourself.

## Scenarios

- **Precondition:**
  - The user is logged in to the app.
  - The plants layer is selected.

>   - At least one plant is placed

Reviewer: ```suggestion
  - At least one plant is placed.
```

- **Main success scenario:**
  - The user adds or removes a plant group from her list of favorites.

> - **Alternative scenario:**

Reviewer 2: Currently, there is no alternative scenario listed. I have a few suggestions:

- The user wants to reorder their list of favorite plant groups
- The user wants to reset their list of favorite plant groups to the default settings

You can dismiss them if you think that is just bloat.

Reviewer: Good idea, there definitely should be a removal of favorites.

> - **Error scenario:**

Reviewer 2: You should consider listing possible error scenarios that may occur. What I thought of for example:
 - The user attempts to add a plant group that is already in their favorites.
 - The user tries to remove a plant group that is not in their favorites.

Reviewer: @NurramoX thank you for the review! Great suggestion! I agree that double-adding shouldn't work.

But: How can one remove something that is not in the list?

Reviewer 2: magine a user has two instances of the application open in separate tabs, and coincidentally, they are on the same page in both tabs where group deletion is possible. In the first tab, the user can delete their favorite group, complete some tasks, close the tab, and then switch to the second tab. If the application does not have a hot-reloading feature, the favorite group will still appear in the second tab despite being deleted in the first one.

Sorry for the confusion

Reviewer: Good point!

- **Postcondition:**
  - The set of map specific favorites has changed according to the changes the user made.  

>     The state of these favorites can be seen in the map view.

Reviewer: What is a map view? (Maybe add to glossary if we actually want/need it.)

```suggestion
    The state of these favorites can be seen in the plant layer.
```

> - **Non-functional Constraints:**

Author: See https://github.com/ElektraInitiative/PermaplanT/pull/168/commits/85cd8c3e6a2357ac6df4b66dc077af752bb0d1f4

# Use Case: Map-Specific Favorite Groups

## Summary

- **Scope:** Plants Layer
- **Level:** User Goal
- **Actors:** App User
- **Brief:** The user changes a set of favorite plant groups per map.
- **Status:** Assigned
- **Assignee:** Benjamin

## Scenarios

- **Precondition:**
  - The user is logged in to the app.
  - The plants layer is selected.
  - At least one plant is placed.
- **Main success scenario:**

>   - The user adds or removes a plant group from her list of favorites.

Reviewer: Can you please also add to this use case that single plant can be added to favorites?

```suggestion
  - The user adds a plant or a plant group to her list of favorites.
```

- **Alternative scenario:**
  - The user wants to reorder their list of favorite plant groups.

>   - The user wants to reset their list of favorite plant groups.

Reviewer: ```suggestion
  - The user wants to remove favorite plant groups.
```

- **Error scenario:**
  - The user attempts to add a plant group that is already in their favorites.
- **Postcondition:**

>   - The set of map specific favorites has changed according to the changes the user made.  

Reviewer: ```suggestion
  - The set of map-specific favorites has changed according to the changes the user made.  
```
